### PR TITLE
Merge master into gcs-nio

### PR DESCRIPTION
**Pull request by jart**
Reviewer: @jean-philippe-martin 
CC: @mziccard

### CODE REVIEW

File: gcloud-java-compute/src/main/java/com/google/cloud/compute/RegionId.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  private RegionId(RegionId regionId) {
+    super(regionId.project());
+    this.region = checkNotNull(regionId.region());
+  }
+
+  /**
+   * Returns the name of the region.
+   */
+  public final String region() {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/cloud/compute/RegionId.java]: Codacy Issue found: [Unnecessary final modifier in final class](https://www.codacy.com/app/mziccard/gcloud-java/file/2745817047/issues/source?bid=3123107&fileBranchId=3356751#l65)

File: gcloud-java-compute/src/main/java/com/google/cloud/compute/InstanceInfo.java [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+          public AttachedDisk apply(AttachedDisk attachedDisk) {
+            return attachedDisk.setProjectId(projectId);
+          }
+        }));
+    return builder.instanceId(instanceId.setProjectId(projectId))
+        .machineType(machineType.setProjectId(projectId))
+        .build();
+  }
+
+  Instance toPb() {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/cloud/compute/InstanceInfo.java]: Codacy Issue found: [The method toPb() has an NPath complexity of 1024](https://www.codacy.com/app/mziccard/gcloud-java/file/2745817030/issues/source?bid=3123107&fileBranchId=3356751#l608)

**Comment by jean-philippe-martin** [comment]
gcloud-java-contrib/gcloud-java-nio-examples/README.md:25 should read -0.2.4-SNAPSHOT instead of -0.2.1.-SNAPSHOT. Same for line 30. The release script updates this one, but of course this will only work once we're merged into master.

**Comment by jean-philippe-martin** [comment]
gcloud-java-examples/README.md we'll want to update the instructions for `ComputeExample` to use appassembler. Not sure whether this needs to be in this PR or if it's better to leave it for a subsequent one, since this PR's already huge.

**Comment by jean-philippe-martin** [comment]
The CL's huge and I can't really check everything, but it looks to me like there's nothing else I'd change, everything else is good.

File: gcloud-java-compute/src/main/java/com/google/cloud/compute/ImageInfo.java [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+    private Long creationTimestamp;
+    private ImageId imageId;
+    private String description;
+    private ImageConfiguration configuration;
+    private Status status;
+    private Long diskSizeGb;
+    private List<LicenseId> licenses;
+    private DeprecationStatus<ImageId> deprecationStatus;
+
+    BuilderImpl() {}
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/cloud/compute/ImageInfo.java]: Codacy Issue found: [Document empty constructor](https://www.codacy.com/app/mziccard/gcloud-java/file/2746909890/issues/source?bid=3123107&fileBranchId=3356751#l149)

File: gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java [hunk 4]
```diff
@@ -0,0 +1,846 @@
...
+
+  com.google.api.services.compute.model.Operation toPb() {
+    com.google.api.services.compute.model.Operation operationPb =
+        new com.google.api.services.compute.model.Operation();
+    if (generatedId != null) {
+      operationPb.setId(new BigInteger(generatedId));
+    }
+    operationPb.setName(operationId.operation());
+    operationPb.setClientOperationId(clientOperationId);
+    switch (operationId.type()) {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java]: Codacy Issue found: [Switch statements should have a default label](https://www.codacy.com/app/mziccard/gcloud-java/file/2746909904/issues/source?bid=3123107&fileBranchId=3356751#l793)

File: gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java [hunk 5]
```diff
@@ -0,0 +1,846 @@
...
+      return true;
+    }
+    if (obj == null || !obj.getClass().equals(Operation.class)) {
+      return false;
+    }
+    Operation other = (Operation) obj;
+    return Objects.equals(toPb(), other.toPb()) && Objects.equals(options, other.options);
+  }
+
+  com.google.api.services.compute.model.Operation toPb() {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java]: Codacy Issue found: [The method toPb() has an NPath complexity of 1024](https://www.codacy.com/app/mziccard/gcloud-java/file/2746909904/issues/source?bid=3123107&fileBranchId=3356751#l785)

File: gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java [hunk 6]
```diff
@@ -0,0 +1,846 @@
...
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(code, message, metadata);
+    }
+  }
+
+  static final class Builder {
+
+    private Compute compute;
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java]: Codacy Issue found: [Private field 'compute' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2746909904/issues/source?bid=3123107&fileBranchId=3356751#l302)

**Comment by jean-philippe-martin** [comment]
Looks good to me.

**Pull request merged**